devtools/x-core/src/lib.rs

```diff
@@ -87,7 +91,7 @@ impl XCoreContext {
     }
 
     /// Returns the Git CLI for this workspace.
-    pub fn git_cli(&self) -> &GitCli {
+    pub fn git_cli(&self) -> &Option<GitCli> {
```

Comment:
This should be `Option<&GitCli>`.

Reply:
(In general, `Option<&T>` is always preferable to `&Option<T>`. You can go from the latter to the former through `.as_ref()`.)

Reply:
Hmm I actually think this should probably return a `Result` type with a friendly error message if it fails, so in callers it can just be bubbled up with `?`.